x-attoparsec/src/X/Data/Attoparsec/Text.hs

```diff
@@ -31,8 +31,7 @@ positiveIntParser :: Parser Int
 positiveIntParser =
   (P.readMaybe <$> many1 digit) >>= P.maybe (fail "not a positive int") pure
 
--- | create a Parser a from a text Parser
-eitherText :: (T.Text -> Either T.Text a) -> Parser a
-eitherText p = do
-  t <- takeText
-  either fail pure (first T.unpack (p t))
+-- | create a Parser a from a text Parser and an input text
+--   usage: takeWhile (/= '-') >>= eitherText parseTimestamp
+eitherText :: (T.Text -> Either T.Text a) -> T.Text -> Parser a
+eitherText p = either (fail . T.unpack) pure . p
```

Comment:
I agree with @charleso 

i mean im not opposed to having both, "to each his own",

But i see more value in the `Either T.Text a -> Parser a`, its more generic in that it doesnt assume an input type of `T.Text`, for instance i can use it witha  function like this

``` Haskell
positiveInt :: Int -> Either T.Text Int
positiveInt x
    | x > 0 = pure x
    | otherwise = Left $ T.concat [(T.pack . show) x, " <= 0"]
```

and it embodies the general concept more, its injecting the "Initial" error monad `Either` into some less minimal monad type with error capabilities